gen/flights/flightplan.py

```diff
@@ -696,6 +703,73 @@ def mission_departure_time(self) -> timedelta:
         return self.sweep_end_time
 
 
+@dataclass(frozen=True)
+class SupportFlightPlan(FlightPlan):
```

Comment:
I think instead of this what you want is a single `AwacsFlightPlan` that subclasses `LoiterFlightPlan` instead of patrol.